### DISCUSSION TITLE
What does [1, 512] tensor of features represent?

### DISCUSSION THREAD

**Discussion by tejask-git** [mikel-brostrom/yolo_tracking#899] [Q&A]
The method _get_features returns a [1, 512] tensor of the detected object's cropped, resized image - https://github.com/mikel-brostrom/yolov8_tracking/blob/737b6ae62b206f718781d3de2398859010e6dcd9/boxmot/strongsort/strong_sort.py#L53

Is it used to make sure that the detected object doesn't move rapidly, or goes out of frame, etc. while it is being tracked?
What does that [1, 512] tensor represent?

**Comment by mikel-brostrom** [comment]
They are embeddings. A learned representation of appearances, where appearances that are similar, have similar representations.

**Reply by tejask-git** [reply]
Thank you @mikel-brostrom 

I tried the [track.py](https://github.com/mikel-brostrom/yolov8_tracking/blob/master/examples/track.py) with an image having a single object and it worked perfectly fine.

I tried to create a model with the reid_export.py script and tried to use that model for the same image.
My model didn't work. When I checked the [1, 512] tensor, it had only one number repeating 512 times. Something like this:
[ 1.797026   1.797026   1.797026   1.797026   1.797026   1.797026   1.797026   1.797026 ... 1.797026   1.797026 ]

I think I could be doing this part wrong - https://github.com/mikel-brostrom/yolov8_tracking/blob/737b6ae62b206f718781d3de2398859010e6dcd9/boxmot/deep/reid_multibackend.py#L43 where we crop and pre-process the detected object.
